src/boilerplate.c

```diff
@@ -443,16 +443,15 @@ void ensure_has_primary_selection() {
 #ifdef HAVE_GTK_PRIMARY_SELECTION
     if (gtk_primary_selection_device_manager != NULL) {
         return;
+    } else {
+        bail("Primary selection is not supported on this compositor");
```

Comment:
No, this is specifically meant to bail if _none_ of the selection protocols (of those supported by wl-clipboard) are supported by the compositor in use, not if _any one_ protocol is unsupported.